src/modules/did/did.service.ts

```diff
@@ -495,4 +559,102 @@ export class DIDService implements OnModuleInit, OnModuleDestroy {
       this.logger.debug(inspect(jobsCounts, { depth: 2, colors: true }));
     }
   }
+
+  /**
+   * Finds DID's changed in given block interval. If fetching of DID's fails at some block, then further fetching stops.
+   *
+   * Returns last successfully synced block
+   */
+  private async getChangedIdentities(
+    fromBlock: number,
+    topBlock: number
+  ): Promise<{ changedIdentities: string[]; syncedBlock: number }> {
+    const didEventFilters = [
+      this.didRegistry.filters.DIDAttributeChanged(null),
+      this.didRegistry.filters.DIDDelegateChanged(null),
+      this.didRegistry.filters.DIDOwnerChanged(null),
+    ];
+    const events = [];
+    let syncedBlock = fromBlock;
+    while (fromBlock < topBlock) {
+      const toBlock = Math.min(
+        topBlock,
+        fromBlock + this.MAX_EVENTS_QUERY_INTERVAL
+      );
+      try {
+        const intervalEvents = (
+          await Promise.all(
+            didEventFilters.map((filter) =>
+              this.didRegistry.queryFilter(filter, fromBlock, toBlock)
+            )
+          )
+        ).flat();
+        this.logger.debug(
+          `Fetched ${intervalEvents.length} DID events from interval [${fromBlock}, ${toBlock}]`
+        );
```

Comment:
Relating to this log line, I've opened this PR: https://github.com/energywebfoundation/ssi-hub/pull/675